faunadb-android/src/test/java/com/faunadb/client/AndroidClientSpec.java

```diff
@@ -51,4 +68,60 @@ public void shouldAuthenticateSession() throws Exception {
     assertThat(identified.to(BOOLEAN).get(), is(false));
   }
 
-}
\ No newline at end of file
+  @Before
+  public void setUpClient() throws Exception {
+    if (rootClient != null)
+      return;
+
+    rootClient = createFaunaClient(ROOT_TOKEN);
+
+    client = transform(
+      transformAsync(
+        setupDatabase(DB_REF, DB_NAME),
```

Comment:
I like where we're going with this.

A few more considerations. I would not put the methods to setup database and keys on the `DslSpec`. It creates this weird cycle where you need to go up to see the schema definition and then, in order to execute the queries, it relies on a method that is defined down on the hierarchy.

If we split out schema definition we could:
- Create the clients on a static `@BecoreClass`
- Move the `@Before` up to te `DslSpec`, making it more or less self contained since all it needs is declared within the class itself. The only down dependency becomes the implementation of `query` method.

To get around the fact that each client is a different class, the schema definition could return expressions that are executed on each specific driver test.

What do you think?